jetty-embedded-11/src/main/java/org/jboss/arquillian/container/jetty/embedded_11/ArquillianAppProvider.java

```diff
@@ -96,10 +98,18 @@ protected App createApp(final Archive<?> archive) {
 
         final File exported;
         try {
-            // If this method returns successfully then it is guaranteed that:
-            // 1. The file denoted by the returned abstract pathname did not exist before this method was invoked, and
-            // 2. Neither this method nor any of its variants will return the same abstract pathname again in the current invocation of the virtual machine.
-            exported = File.createTempFile(EXPORT_FILE_PREFIX, archive.getName(), EXPORT_DIR);
+            if (this.config.isUseArchiveNameAsContext()) {
+                Path tmpDirectory = Files.createTempDirectory("arquillian-jetty");
+                Path archivePath = tmpDirectory.resolveSibling(archive.getName());
+                Files.deleteIfExists(archivePath);
+                exported = Files.createFile(archivePath).toFile();
+                exported.deleteOnExit();
+            } else {
+                // If this method returns successfully then it is guaranteed that:
+                // 1. The file denoted by the returned abstract pathname did not exist before this method was invoked, and
+                // 2. Neither this method nor any of its variants will return the same abstract pathname again in the current invocation of the virtual machine.
+                exported = File.createTempFile(EXPORT_FILE_PREFIX, archive.getName(), EXPORT_DIR);
```

Comment:
Ditto, same else branch

Reply:
I know there is a lot of code duplicated for each jetty 9/10/11 but I don't really want to touch this now.